addons/medical_vitals/functions/fnc_updateOxygen.sqf

```diff
@@ -0,0 +1,83 @@
+#include "script_component.hpp"
+/*
+ * Author: Brett Mayson
+ * Update the oxygen levels
+ *
+ * Arguments:
+ * 0: The Unit <OBJECT>
+ * 1: Time since last update <NUMBER>
+ * 2: Sync value? <BOOL>
+ *
+ * ReturnValue:
+ * Current SPO2 <NUMBER>
+ *
+ * Example:
+ * [player, 1, false] call ace_medical_vitals_fnc_updateOxygen
+ *
+ * Public: No
+ */
+
+params ["_unit", "_deltaT", "_syncValue"];
+
+#define IDEAL_PPO2 0.255
+
+private _current = GET_SPO2(_unit);
+private _heartRate = GET_HEART_RATE(_unit);
+
+private _altitude = (getPosASL _unit) select 2;
+private _po2 = if (missionNamespace getVariable ["ace_weather", false]) then {
+    private _temperature = _altitude call EFUNC(weather,calculateTemperatureAtHeight);
+    private _pressure = _altitude call EFUNC(weather,calculateBarometricPressure);
+    [_temperature, _pressure, EGVAR(weather,currentHumidity)] call EFUNC(weather,calculateOxygenDensity)
+} else {
+    // Rough approximation of the partial pressure of oxygen in the air
+    0.25725 * (_altitude / 1000 + 1)
+};
+
+private _oxygenSaturation = (IDEAL_PPO2 min _po2) / IDEAL_PPO2;
+
+// Check gear for oxygen supply
+[goggles _unit, headgear _unit] findIf {
+    if (_x != "") then {
+        private _condition = getText (configFile >> "CfgWeapons" >> _x >> QGVAR(oxygenSupply));
+        if (_condition != "" && {ace_player call compile _condition}) then {
```

Comment:
IMO these should be cached if this is running out of the statemachine.

Reply:
Yeah, I just didn't bother with it yet, meant to leave a comment but forgot as tweaking took way longer than expected